core/tests/vm-benchmark/harness/src/lib.rs

```diff
@@ -96,33 +181,62 @@ impl BenchmarkingVm {
                 default_validation_computational_gas_limit: BATCH_COMPUTATIONAL_GAS_LIMIT,
                 chain_id: L2ChainId::from(270),
             },
-            &*STORAGE,
+            &STORAGE,
         ))
     }
+}
 
+impl<VM: BenchmarkingVmFactory> BenchmarkingVm<VM> {
     pub fn run_transaction(&mut self, tx: &Transaction) -> VmExecutionResultAndLogs {
         self.0.push_transaction(tx.clone());
         self.0.execute(VmExecutionMode::OneTx)
     }
 
+    pub fn run_transaction_full(&mut self, tx: &Transaction) -> VmExecutionResultAndLogs {
+        self.0.pop_snapshot_no_rollback();
```

Comment:
I'm not a fan of pop, then snapshot like I already stated in the sequencer. I think it is more sensible to snapshot, then pop at the end so there is no unnecessary pop on the first execution.